plotters/src/element/pie.rs

```diff
@@ -135,9 +152,18 @@ impl<'a, DB: DrawingBackend, Label: Display> Drawable<DB> for Pie<'a, (i32, i32)
                 points.push(coord);
                 offset_theta += radian_increment;
             }
+            if self.donut_hole > &0.0 {
+                while offset_theta >= slice_start {
+                    let coord = theta_to_ordinal_coord(*self.donut_hole, offset_theta, self.center);
+                    points.push(coord);
+                    offset_theta -= radian_increment;
+                }
+            }
             // final point of the wedge may not fall exactly on a radian, so add it extra
-            let final_coord = theta_to_ordinal_coord(*self.radius, theta_final, self.center);
-            points.push(final_coord);
+            if self.donut_hole == &0.0 {
+                let final_coord = theta_to_ordinal_coord(*self.radius, theta_final, self.center);
+                points.push(final_coord);
+            }
```

Comment:
I'm not entirely sure whether that's the best approach. The idea behind the final point was to make it possible to hit the position exactly, even if no increment of `radian_increment` fits in the remaining space. I would keep this unconditional and would start subtracting from `final_coord` on the way back and then calculate a second `final_coord` for the final inner edge.